storage/storage_transport_test.go

```diff
@@ -0,0 +1,137 @@
+package storage
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/containers/image/docker/reference"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	sha256digestHex = "0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef"
+)
+
+func TestTransportName(t *testing.T) {
+	assert.Equal(t, "containers-storage", Transport.Name())
+}
+
+func TestTransportParseStoreReference(t *testing.T) {
+	for _, c := range []struct{ input, expectedRef, expectedID string }{
+		{"", "", ""}, // Empty input
+		// Handling of the store prefix
+		// FIXME? Should we be silently discarding input like this?
+		{"[unterminated", "", ""},                                    // Unterminated store specifier
+		{"[garbage]busybox", "docker.io/library/busybox:latest", ""}, // Store specifier is ignored
+
+		{"UPPERCASEISINVALID", "", ""},                                                     // Invalid single-component name
+		{"sha256:" + sha256digestHex, "docker.io/library/sha256:" + sha256digestHex, ""},   // Valid single-component name
+		{"@" + sha256digestHex, "", sha256digestHex},                                       // Valid single-component ID
+		{"sha256:ab", "docker.io/library/sha256:ab", ""},                                   // Valid single-component name (ParseIDOrReference accepts digest prefixes as names!) (FIXME? is this desirable?)
```

Comment:
Now that IDs are unambiguously identified using @, the comment for this case can drop the FIXME and a reference to ParseIDOrReference which is no longer used, and just casually mention that this looks like a digest prefix but without the @ clearly isn’t one, or something like that.